src/collections/_documentation/apm/distributed-tracing.md

```diff
@@ -0,0 +1,344 @@
+    span.set_data("http.foobarsessionid", get_foobar_sessionid())
+```
+
+### JavaScript
+
+To access our APM features, you need to use a beta release.
+
+```bash
+$ npm install @sentry/browser
+$ npm install @sentry/apm
+```
+
+**Sending Traces**
+
+Tracing resides in the `@sentry/apm` package. You can add it to your `Sentry.init` call:
+
+```javascript
+import * as Sentry from '@sentry/browser';
+import { Integrations as ApmIntegrations } from '@sentry/apm';
+
+Sentry.init({
+  dsn: '___PUBLIC_DSN___',
+  integrations: [
+    new ApmIntegrations.Tracing(),
+  ],
+});
+```
+
+You can pass a lot of different options to tracing, but it comes with reasonable defaults out of the box. You'll receive:
+
+- A [transaction]({%- link _documentation/apm/apm-glossary.md -%}#transaction) for every page load
+- All XHR/fetch requests as spans
+
+**Using Tracing Integration for Manual Tracing**
+
+Tracing will create a [transaction]({%- link _documentation/apm/apm-glossary.md -%}#transaction) on page load by default; all [spans]({%- link _documentation/apm/apm-glossary.md -%}#span) that are started will be attached to it. Also, tracing will finish the transaction after the default timeout of 500ms of inactivity. Inactivity is if there are no pending XHR/fetch requests. If you want to extend it by adding spans to the transaction, here is an example of how you could profile a React component:
+
+```javascript
+// This line starts an activity (and creates a span).
+// As long as you don't pop the activity, the transaction will not be finished and therefore not sent to Sentry.
+// If you do not want to create a span out of an activity, just don't provide the second arg.
+const activity = ApmIntegrations.Tracing.pushActivity(displayName, {
+        data: {},
+        op: 'react',
+        description: `<${displayName}>`,
+});
+
+// Sometime later ...
+// When we pop the activity, the Integration will finish the span and after the timeout finish the transaction and send it to Sentry
+Integrations.ApmIntegrations.popActivity(activity);
+```
+
+**Managing Transactions**
+
+Let’s say you want to create a [transaction]({%- link _documentation/apm/apm-glossary.md -%}#transaction) for an expensive operation (for example, `processItem`) and send the result to Sentry:
+
+```javascript
+const transaction = Sentry.getCurrentHub().startSpan({ op: "task",  transaction: item.getTransaction() })
+  // processItem may create more spans internally (see next example)
+  processItem(item).then(() => {
+    transaction.finish()
+  })
+```
+
+**Adding Additional Spans to the Transaction**
+
+The next example is called in the `processItem` function from the code snippet above. Our SDK can determine if there is a current open `transaction` and add all newly created [spans]({%- link _documentation/apm/apm-glossary.md -%}#span) as child operations to the `transaction`. Keep in mind; each individual span also needs to be finished; otherwise, it will not show up in the `transaction`.
```

Comment:
```suggestion
The next example contains the implementation of the hypothetical `processItem ` function called from the code snippet in the previous section. Our SDK can determine if there is a current open `transaction` and add all newly created [spans]({%- link _documentation/apm/apm-glossary.md -%}#span) as child operations to the `transaction`. Keep in mind that each individual span also needs to be manually finished; otherwise, spans will not show up in the `transaction`.
```